File: backend/src/main/java/com/dhairya/Placement_management_system/drive/Drive.java
package com.dhairya.Placement_management_system.drive;

import com.dhairya.Placement_management_system.user.User;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "drives")
@Getter
@Setter
@NoArgsConstructor
public class Drive {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "min_gpa", precision = 3, scale = 2)
    private BigDecimal minGpa;

    @Column(name = "allowed_graduation_years")
    private Integer[] allowedGraduationYears;

    @Column(name = "required_skills")
    private String[] requiredSkills;

    @Column(name = "additional_criteria", columnDefinition = "TEXT")
    private String additionalCriteria;

    @Column(name = "application_deadline", nullable = false)
    private LocalDateTime applicationDeadline;

    @Column(name = "drive_date")
    private LocalDateTime driveDate;

    @Column(nullable = false)
    private String status = "DRAFT";

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by", nullable = false)
    private User createdBy;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/drive.ts
import type { UserResponse } from './user'

export interface CreateDriveRequest {
  title: string
  description?: string
  minGpa?: number | null
  allowedGraduationYears?: number[]
  requiredSkills?: string[]
  additionalCriteria?: string
  applicationDeadline: string
  driveDate?: string | null
}

export interface UpdateDriveRequest {
  title?: string
  description?: string
  minGpa?: number | null
  allowedGraduationYears?: number[]
  requiredSkills?: string[]
  additionalCriteria?: string
  applicationDeadline?: string
  driveDate?: string | null
}

export interface DriveResponse {
  id: string
  title: string
  description: string | null
  minGpa: number | null
  allowedGraduationYears: number[] | null
  requiredSkills: string[] | null
  additionalCriteria: string | null
  applicationDeadline: string
  driveDate: string | null
  status: 'DRAFT' | 'ACTIVE' | 'CLOSED' | 'COMPLETED'
  createdBy: UserResponse
  createdAt: string
  updatedAt: string
}
